lib/improver/utilities/cube_metadata.py

```diff
@@ -252,7 +252,7 @@ def update_coord(cube, coord_name, changes, warnings_on=False):
                        " coord {}".format(coord_name))
                 raise ValueError(msg)
         if 'bounds' in changes:
-            new_bounds = np.array(changes['bounds'])
+            new_bounds = np.array(changes['bounds'], dtype=new_coord.dtype)
```

Comment:
I have a feeling we might want to get rid of this function as part of this theme of work (although probably not this ticket).  What do you think?

Reply:
I don't think we need to get rid of it, just ensure that non-conformant changes are not allowed. Effectively add an additional check at the end of these functions to ensure any non-compliant change results in an error.